rs/nns/integration_tests/src/neuron_following.rs

```diff
@@ -537,6 +539,153 @@ async fn test_prune_some_following() {
+        );
+    }
+
+    let voting_power_refreshed_timestamp_seconds_neuron_values = vec![
+        EVIL_TIMESTAMP_SECONDS - 1,
+        EVIL_TIMESTAMP_SECONDS,
+        EVIL_TIMESTAMP_SECONDS + 1,
+        now_seconds,
+    ];
+
+    let cached_neuron_stake_e8s_neuron_values = vec![
+        0,       // inactive -> stable memory
+        42 * E8, // active -> heap
+    ];
+
+    let original_neurons = voting_power_refreshed_timestamp_seconds_neuron_values
+        .into_iter()
+        .cartesian_product(cached_neuron_stake_e8s_neuron_values.into_iter())
+        .collect::<Vec<_>>();
+
+    let controller = PrincipalId::new_user_test_id(482_783_461);
+
+    let ids = 100..;
+    let original_neurons = ids
+        .zip(original_neurons.into_iter())
+        .map(
+            |(id, (voting_power_refreshed_timestamp_seconds, cached_neuron_stake_e8s))| Neuron {
+                id: Some(NeuronId { id }),
+                controller: Some(controller),
+                account: vec![id as u8; 32],
+                cached_neuron_stake_e8s,
+                voting_power_refreshed_timestamp_seconds: Some(
+                    voting_power_refreshed_timestamp_seconds,
+                ),
+                ..neuron_base.clone()
+            },
+        )
+        .collect::<Vec<_>>();
+
+    assert_eq!(original_neurons.len(), 8);
+
+    let governance_proto = GovernanceCanisterInitPayloadBuilder::new()
+        .with_additional_neurons(original_neurons.clone())
+        .build();
+
+    install_canister(
+        &pocket_ic,
+        "NNS Governance",
+        GOVERNANCE_CANISTER_ID,
+        governance_proto.encode_to_vec(),
+        // TODO: Once following pruning is released, replace with vanilla
+        // build_governance_wasm(). For now, the feature is only enabled when
+        // built with feature = "test".
+        build_test_governance_wasm(),
+        Some(ROOT_CANISTER_ID.get()),
+    )
+    .await;
+
+    // Step 2: Call the code under test.
+
+    // Wait for backfilling to complete.
+    for _ in 0..100 {
+        pocket_ic.advance_time(Duration::from_secs(1)).await;
+        pocket_ic.tick().await;
+    }
+
+    // Step 3: Inspect results
+
+    let mut observed_neurons = nns::governance::list_neurons(&pocket_ic, controller)
+        .await
+        .full_neurons;
+    assert_eq!(observed_neurons.len(), 8);
+    observed_neurons.sort_by_key(|neuron| neuron.id.as_ref().unwrap().id);
+
+    assert_eq!(
+        observed_neurons
+            .iter()
+            .map(|neuron| neuron.voting_power_refreshed_timestamp_seconds.unwrap())
+            .collect::<Vec<u64>>(),
+        // Expected values being repeated repeated reflects the fact that we are
+        // testing both heap and stable memory neurons.
+        vec![
+            EVIL_TIMESTAMP_SECONDS - 1,
+            EVIL_TIMESTAMP_SECONDS - 1,
+            DEFAULT_VOTING_POWER_REFRESHED_TIMESTAMP_SECONDS,
+            DEFAULT_VOTING_POWER_REFRESHED_TIMESTAMP_SECONDS,
```

Comment:
This shows that the desired changes were made.

(The other lines show that no undesired changes were made to voting_power_refreshed_timestamp_seconds.)